Peter

petermcclurg opened the pull request:
Pull request to assess census and bed days script for discussion Monday

petermcclurg committed:
previous version of census and bed days

petermcclurg committed:
pulled code from master set up environment

Merge remote-tracking branch 'origin/master' into peter

# Please enter a commit message to explain why this merge is necessary,
# especially if it merges an updated upstream into a topic branch.
#
# Lines starting with '#' will be ignored, and an empty message aborts
# the commit.

petermcclurg committed:
updated variable names. data matching spss output for feb 2020. needs environment automatic update to master version. needs updated filepaths within environment to save adding in script

petermcclurg committed:
recommit test

petermcclurg committed:
fix typo

petermcclurg committed:
commit update

petermcclurg committed:
test commit

petermcclurg committed:
file move for update

petermcclurg committed:
amended function still needs checking re: bringing in validated file

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+##########################################################
+# Name of file: 04_Census_and_Bed_Days.R
```
```suggestion
# Name of file: 03_census_and_bed_days.R
```

alice-hannah replied:
I'd actually leave this as number 4 as we will move the script to create the trend file to be script 3. Agree to change to lower case though.

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+##########################################################
+# Name of file: 04_Census_and_Bed_Days.R
+# Data Release: Delayed Discharges monthflagly publication
```
```suggestion
# Data release: Monthly Delayed Discharges publication
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+# Name of file: 04_Census_and_Bed_Days.R
+# Data Release: Delayed Discharges monthflagly publication
+# Original author(s): Peter McClurg (spss version: James Mc Nally)
+# Original date: 09/10/19 (spss version: 30/11/2017)
```
Don't need to have the callbacks to SPSS, you're the author now ;)
```suggestion
# Original author(s): Peter McClurg
# Original date: August 2020
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+# Original date: 09/10/19 (spss version: 30/11/2017)
+# latest update author (if not using version control)
+# latest update date (if not using version control)
+# latest update description (if not using version control)
```
You're using version control, so let's remove all the unnecessary commentary.
```suggestion
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+# latest update author (if not using version control)
+# latest update date (if not using version control)
+# latest update description (if not using version control)
+# Type of script: Preparation
```
```suggestion
# Type: Reproducible Analytical Pipeline (RAP)
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+# latest update description (if not using version control)
+# Type of script: Preparation
+# Written/run on: R Studio SERVER
+# Version of R that the script was most recently run on: ?
```
This is shown in the top right-hand corner, under log-out and sessions, on the server. If you're using the latest version available, that's `3.6.1`
```suggestion
# Version of R that the script was most recently run on: 3.6.1
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+#   setting filepaths and extract dates
+#   functions (defined here or sourced from another file)
+#   setting plot parameter
+#   specifying codes (e.g. ICD-10 codes)
```
No "housekeeping" should be done in the separate script files, this is all handled in `setup_environment.R`. Removing this as a section will have a knock-on effect for section numbering.
```suggestion
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+### Filepaths for latest monthflag # may NOT NEEDED as now within set-up_environment ( check with Russell)
+
+filepath<-("/conf/delayed_discharges/RAP development/2020_02/Outputs/")
+filepath2<-("/conf/delayed_discharges/RAP development/2020_02/Data/scotland/")
```
You're right, these aren't needed. They are also written as "absolute" file paths (from `/conf/`) so this wouldn't work on different systems (e.g. desktop/server). The project structure is set up so that all directories and files were held within the project, this means we can use "relative" (based on where the project is, not the system) as opposed to "absolute". I'll comment on any suggestions for how to handle this as needed.
```suggestion
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+
+### 2. Bring in environment ----
+#( ran separately in code folder - how do i bring it in without copying / pasting as ?)
+source("00_setup_environment.R")
```
I think we went-over loading the `setup_environment` file which wasn't working with `source()` because `census_and_bed_days` wasn't in the code folder. You've moved it now, so this line should work fine. However, by using `here()`, it adds some reassurance that across systems and users, the file paths should remain stable.

We'll also add in the loading of the functions using the same method Alice uses in `create_scotland_file`.

```suggestion
### 0 - Load setup environment and functions ----
# Setup environment
source(here::here("code", "00_setup_environment.R"))

# Functions
walk(list.files(here("functions"), full.names = TRUE), source)
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+source("00_setup_environment.R")
+
+#add in new library needed that isn't in setup.
+library(xlsx)
```
Loading packages shouldn't be done in individual scripts. If there is a package you need, add it to the `setup_environment` file (then commit/push/pull request). I believe we'd recommend the use of `openxlsx` instead of `xlsx` though.

```suggestion
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+#add in new library needed that isn't in setup.
+library(xlsx)
+
+# monthflagflag<-("Apr 2020")
```
Is this required?
```suggestion
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+
+# monthflagflag<-("Apr 2020")
+
+### 3. Get Scotland_validated file for latest monthflag ----
```
```suggestion
### 1 - Import Scotland_validated file ----
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+
+### 3. Get Scotland_validated file for latest monthflag ----
+
+datafile<-read.csv(paste0(filepath2,"SCOTLAND_validated.csv"))
```
This file doesn't exist and isn't created from previous scripts? Assuming it's the output from `02_create_scotland_file.R`, the method to read in that file without using a hardcoded file path is suggested below.
```suggestion
datafile <- readRDS(here("data", format(start_month, "%Y-%m"), paste0(format(start_month, "%Y-%m"), "_scotland.rds")))
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+### 3. Get Scotland_validated file for latest monthflag ----
+
+datafile<-read.csv(paste0(filepath2,"SCOTLAND_validated.csv"))
+#sum(datafile$obds_in_month) check this matches publication when testing prevous data.
```
See "Exploration/Checking Statements" section in general feedback comments.
```suggestion
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+#strip out code 100s
+
+
+datafile<-datafile %>% filter(datafile$reas1!="Code 100")
```
See "Pipes" section in general feedback comments.
```suggestion
# Filter to remove 'Code 100'
datafile <- datafile %>% filter(datafile$reas1 != "Code 100")
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+
+
+datafile<-datafile %>% filter(datafile$reas1!="Code 100")
+#table(datafile$reas1) # check code 100 has been removed
```
See "Exploration/Checking Statements" section in general feedback comments.
```suggestion
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+#Create new variables
+
+datafile<-datafile %>% mutate(areaname=" ")
+datafile<-datafile %>% mutate(year="2019-20")
```
**Double-check**: don't think `areaname` needs to be defined and think `year` is defined in a previous script.
```suggestion
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+#          discharge_date=DateDischarge,
+#          discharge_to_code=DischargeReason,
+#          census_flag=CENSUSFLAG,
+#          delay_description=DELAY_DESCRIPTION)
```
I believe some of this, if not all, is completed as part of `02_create_scotland_file` - my opinion would be to keep these sort of changes there anyway.
```suggestion
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+datafile<-datafile%>% mutate(age_grp=
+                               if_else(AGEATRDD<75, "18-74",
+                                       if_else(AGEATRDD>=75, "75+", " ")))
+
```
This variable grouping is created in `02_create_scotland_file`
```suggestion
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+#save out file for main bed days file
+ 
+
+#write_sav(datafile,paste0(filepath,"main bed days file.sav")) # save out file
```
See "Saving (SPSS) Files Out" section in general feedback comments.
The only change to the original `_scotland.rds` file is the filter for excluding `"Code 100"`.
```suggestion
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+#write_sav(datafile,paste0(filepath,"main bed days file.sav")) # save out file
+
+#set function
+#'%!in%' <- function(x,y)!('%in%'(x,y))
```
This function is commented out and is replaced with code written below. 
```suggestion
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+#set function
+#'%!in%' <- function(x,y)!('%in%'(x,y))
+
+#Save census data file
```
This comment relates to code further down that I suggested is removed.
```suggestion
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+#Save census data file
+
+
+datafile2 <- filter(datafile, !REASONFORDELAYSECONDARY %in%c("26X","46X") & CENSUSFLAG=="Y")
```
See "Incremental Data-Frame Names" section in general feedback comments.
```suggestion
datafile <- filter(datafile, !(reasonfordelaysecondary %in% c("26X", "46X")) & census_flag == 1)
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+datafile2 <- filter(datafile, !REASONFORDELAYSECONDARY %in%c("26X","46X") & CENSUSFLAG=="Y")
+
+
+#write_sav(datafile2,paste0(filepath,"main census file.sav")) # save out file
```
See "Saving (SPSS) Files Out" section in general feedback comments.
```suggestion
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+           if_else(DELAY_DESCRIPTION=="Awaiting availability of transport","Transport",reas2))
+
+
+#write_sav(datafile3,paste0(filepath,"main census file Tabs 1 3 6.sav")) # save out file
```
See "Saving (SPSS) Files Out" section in general feedback comments.
```suggestion
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+
+#write_sav(datafile3,paste0(filepath,"main census file Tabs 1 3 6.sav")) # save out file
+
+### 6. Get reason / age_grp breakdown for Scotland ----
```
```suggestion
### 2 - Get reason / age_grp breakdown for Scotland (Level = 2) ----
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+  mutate(reas2 = as.character(reas2))%>% 
+  mutate(reas2=
+           if_else(DELAY_DESCRIPTION=="Awaiting availability of transport","Transport",reas2))
+
```
This appears to be redundant, the `reas2` column is populated during `create_scotland_file`. 
```suggestion
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+
+### 6. Get reason / age_grp breakdown for Scotland ----
+
+datafile4<-datafile3
```
See "Incremental Data-Frame Names" section in general feedback comments.
```suggestion
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+datafile4<-datafile3
+datafile4$level<-"1"
+
+datafile4$areaname<-"Scotland"
```
See "Style Guide", "Pipes", and "Code Needs Refactoring" sections in general feedback comments.

It may seem arbitrary but our best practice for adding new columns would be to use the `mutate()` function, it's easier to understand and the code is more efficient. However, this is usually used to add columns as a function of existing ones. 
```suggestion
datafile %<>% mutate(level = "1", areaname = "Scotland")
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+
+datafile4$areaname<-"Scotland"
+
+Scotlandbymainreasongrouping <- datafile4 %>% 
+  group_by(year,MONTHFLAG,level,areaname,age_grp,reas1) %>% 
+  summarise(NoofPatients=sum(NoofPatients,na.rm=TRUE),Delay1to3days=sum(Delay1to3days,na.rm=TRUE),
+            Delay3to14days=sum(Delay3to14days,na.rm=TRUE),Delay2to4weeks=sum(Delay2to4weeks,na.rm=TRUE),
+            Delay4to6weeks=sum(Delay4to6weeks,na.rm=TRUE),Delay6to12weeks=sum(Delay6to12weeks,na.rm=TRUE),
+            Delay3to6months=sum(Delay3to6months,na.rm=TRUE),Delay6to12months=sum(Delay6to12months,na.rm=TRUE),
+            DelayOver12months=sum(DelayOver12months,na.rm=TRUE),DelayOver3days=sum(DelayOver3days,na.rm=TRUE),
+            DelayUnder2wks=sum(DelayUnder2wks,na.rm=TRUE),
+            DelayOver6wks=sum(DelayOver6wks,na.rm=TRUE),DelayOver4wks=sum(DelayOver4wks,na.rm=TRUE),
+            DelayOver2wks=sum(DelayOver2wks,na.rm=TRUE), acute=sum(acute,na.rm=TRUE),
+            gpled=sum(gpled,na.rm=TRUE),notgpled=sum(notgpled,na.rm=TRUE)) %>% 
+  ungroup()
```
See "Style Guide" and "Incorrect Variable Names" sections in general feedback comments.

Have a look at the `summarise_at()` function. This will make the code more concise. I'll make a suggestion for this but the same can be done throughout the script where a similar pattern occurs.
```suggestion
Scotlandbymainreasongrouping <- datafile %>%
    group_by(year, MONTHFLAG, level, areaname, age_grp, reas1) %>%
    summarise_at(vars(num_pats, delay_1_to_3_days:delay_over_6_weeks, acute:notgpled),
               ~ sum(., na.rm = TRUE)) %>%
    ungroup()
```

alice-hannah replied:
Agree with this idea - however `summarise_at` has been superseded by `across` in this new version of dplyr so I'd suggest using this instead.

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+            gpled=sum(gpled,na.rm=TRUE),notgpled=sum(notgpled,na.rm=TRUE)) %>% 
+  ungroup()
+
+#write_sav(Scotlandbymainreasongrouping,paste0(filepath,"Scotland by main reason grouping.sav")) # save out file
```
See "Saving (SPSS) Files Out" section in general feedback comments.
```suggestion
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+
+
+
+#write_sav(Scotlandbysubreasongrouping,paste0(filepath,"Scotland by sub reason grouping.sav")) # save out file
```
See "Saving (SPSS) Files Out" section in general feedback comments.
```suggestion
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+
+#write_sav(Scotlandbysubreasongrouping,paste0(filepath,"Scotland by sub reason grouping.sav")) # save out file
+
+### 7. Breakdown for HBs (Level=2) ----
```
```suggestion
### 3 - Breakdown for HBs (Level = 2) ----
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+
+### 7. Breakdown for HBs (Level=2) ----
+datafile4$level<-"2"
+datafile4<-datafile4 %>% mutate(areaname=Healthboard)
```
See "Style Guide", "Pipes", and "Code Needs Refactoring" sections in general feedback comments.

```suggestion
datafile %<>% mutate(level = "2", areaname = "Healthboard")
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+            gpled=sum(gpled,na.rm=TRUE),notgpled=sum(notgpled,na.rm=TRUE)) %>% 
+  ungroup()
+
+#write_sav(HBbymainreasongrouping,paste0(filepath,"HB by main reason grouping.sav")) # save out file
```
See "Saving (SPSS) Files Out" section in general feedback comments.
```suggestion
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+            gpled=sum(gpled,na.rm=TRUE),notgpled=sum(notgpled,na.rm=TRUE)) %>% 
+  ungroup()
+
+#write_sav(HBbysubreasongrouping,paste0(filepath,"HB by sub reason grouping.sav")) # save out file
```
See "Saving (SPSS) Files Out" section in general feedback comments.
```suggestion
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+
+#write_sav(HBbysubreasongrouping,paste0(filepath,"HB by sub reason grouping.sav")) # save out file
+
+### 8. Breakdown for LA ----
```
```suggestion
### 4 - Breakdown for LAs (Level = 3) ----
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+
+### 8. Breakdown for LA ----
+datafile4$level<-"3"
+datafile4<-datafile4 %>% mutate(areaname=LocalAuthorityArea)
```
See "Style Guide", "Pipes", and "Code Needs Refactoring" sections in general feedback comments.

```suggestion
datafile %<>% mutate(level = "3", areaname = LocalAuthorityArea)
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+            gpled=sum(gpled,na.rm=TRUE),notgpled=sum(notgpled,na.rm=TRUE)) %>% 
+  ungroup()
+
+#write_sav(labysubreasongrouping,paste0(filepath,"la by sub reason grouping.sav")) # save out file
```
See "Saving (SPSS) Files Out" section in general feedback comments.
```suggestion
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+#write_sav(labysubreasongrouping,paste0(filepath,"la by sub reason grouping.sav")) # save out file
+
+
+### 9. Add files together ----
```
```suggestion
### 5 - Combine summary dataframes ----
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+### 9. Add files together ----
+
+Scot_HB_la<-bind_rows(Scotlandbymainreasongrouping,Scotlandbysubreasongrouping,HBbymainreasongrouping,HBbysubreasongrouping,
+                      labymainreasongrouping,labysubreasongrouping)
```
See "Style Guide" section in general feedback comments.
```suggestion
Scot_HB_la <- bind_rows(Scotlandbymainreasongrouping, Scotlandbysubreasongrouping, HBbymainreasongrouping, HBbysubreasongrouping, labymainreasongrouping, labysubreasongrouping)
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+
+
+
+#if reas1 is blank(N/A), reas1 = reas2
```
```suggestion
# if reas1 is blank(N/A), reas1 = reas2
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+datafile6<-Scot_HB_la%>% 
+  mutate(reas1 = as.character(reas1))%>% 
+  mutate(reas1=
+           if_else(is.na(reas1), reas2,reas1))
```
See "Style Guide" and "Incremental Data-Frame Names" sections in general feedback comments.
```suggestion
Scot_HB_la %>% 
    mutate(reas1 = as.character(reas1)) %>% 
    mutate(reas1 = if_else(is.na(reas1), reas2, reas1))
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+#remove reason_grp
+datafile6<-select(datafile6,-reas2)
+
+datafile6a<-datafile6
```
See "Incremental Data-Frame Names" and "Code Needs Refactoring" sections in general feedback comments.
```suggestion
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+datafile6<-select(datafile6,-reas2)
+
+datafile6a<-datafile6
+datafile6a$age_grp<-"All"
```
See "Style Guide", "Pipes", "Incremental Data-Frame Names" and "Code Needs Refactoring" sections in general feedback comments.
```suggestion
Scot_HB_la %<>% mutate(age_grp = "All")
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+            gpled=sum(gpled,na.rm=TRUE),notgpled=sum(notgpled,na.rm=TRUE)) %>% 
+  ungroup()
+
+#write_sav(ScotHBlaallage_grps,paste0(filepath,"ScotHBla all age_grps.sav")) # save out file
```
See "Saving (SPSS) Files Out" section in general feedback comments.
```suggestion
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+#write_sav(ScotHBlaallage_grps,paste0(filepath,"ScotHBla all age_grps.sav")) # save out file
+
+
+ScotHBlaallreasonexcHSCPatFamtotal<-bind_rows(datafile6,ScotHBlaallage_grps)
```
See "Style Guide" and "Incremental Data-Frame Names" sections in general feedback comments.
```suggestion
ScotHBlaallreasonexcHSCPatFamtotal <- bind_rows(Scot_HB_la, ScotHBlaallage_grps)
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+
+
+ScotHBlaallreasonexcHSCPatFamtotal<-bind_rows(datafile6,ScotHBlaallage_grps)
+table(ScotHBlaallreasonexcHSCPatFamtotal$reas1) # check on reason_grp_high_level
```
See "Exploration/Checking Statements" section in general feedback.
```suggestion
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+ScotHBlaallreasonexcHSCPatFamtotal<-bind_rows(datafile6,ScotHBlaallage_grps)
+table(ScotHBlaallreasonexcHSCPatFamtotal$reas1) # check on reason_grp_high_level
+
+#write_sav(ScotHBlaallreasonexcHSCPatFamtotal,paste0(filepath,"ScotHBla all reasons exc HSC PatFam total.sav")) # save out file
```
See "Saving (SPSS) Files Out" section in general feedback comments.
```suggestion
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+#write_sav(ScotHBlaallreasonexcHSCPatFamtotal,paste0(filepath,"ScotHBla all reasons exc HSC PatFam total.sav")) # save out file
+
+
+###10. Calculate total number of delays excluding code9s ----
```
```suggestion
### 6 - Calculate the total number of delays excluding where code = 9 ----
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+
+###10. Calculate total number of delays excluding code9s ----
+datafile7<-filter(datafile6, reas1%in%c("Health and Social Care Reasons","Patient/Carer/Family-related reasons"))
+datafile7$reas1<-"All Delays excl. Code 9"
```
See "Style Guide", "Pipes", and "Incremental Data-Frame Names" sections in general feedback comments.
```suggestion
Scot_HB_la %<>% 
    filter(reas1 %in% c("Health and Social Care Reasons", "Patient/Carer/Family-related reasons")) %>%
    mutate(reas1 = "All Delays excl. Code 9")
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+            gpled=sum(gpled,na.rm=TRUE),notgpled=sum(notgpled,na.rm=TRUE)) %>% 
+  ungroup()
+
+#write_sav(ScotHBlaallreasonsincHSCPatFamtotal,paste0(filepath,"ScotHBla all reasons inc HSC PatFam total.sav")) # save out file
```
See "Saving (SPSS) Files Out" section in general feedback comments.
```suggestion
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+#write_sav(ScotHBlaallreasonsincHSCPatFamtotal,paste0(filepath,"ScotHBla all reasons inc HSC PatFam total.sav")) # save out file
+
+
+###11. Calculate the number of delays for all reasons ----
```
```suggestion
###11. Calculate the number of delays for all reasons ----
### 7 - Calculate the number of delays for all reasons ----
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+###11. Calculate the number of delays for all reasons ----
+
+datafile8<-filter(ScotHBlaallreasonexcHSCPatFamtotal, reas1%in%c("Health and Social Care Reasons","Code 9",
+                                                                                 "Patient/Carer/Family-related reasons"))
```
See "Style Guide", "Pipes", and "Incremental Data-Frame Names" sections in general feedback comments.
```suggestion
Scot_HB_la %<>% filter(ScotHBlaallreasonexcHSCPatFamtotal, reas1 %in% c("Health and Social Care Reasons", "Code 9", "Patient/Carer/Family-related reasons"))
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+
+datafile8<-filter(ScotHBlaallreasonexcHSCPatFamtotal, reas1%in%c("Health and Social Care Reasons","Code 9",
+                                                                                 "Patient/Carer/Family-related reasons"))
+table(datafile8$reas1) # check totals against the syntax output 
```
See "Exploration/Checking Statements" section in general feedback.
```suggestion
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+                                                                                 "Patient/Carer/Family-related reasons"))
+table(datafile8$reas1) # check totals against the syntax output 
+
+datafile8$reas1<-"All"
```
```suggestion
Scot_HB_la %>% mutate(reas1 = "All")
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+            gpled=sum(gpled,na.rm=TRUE),notgpled=sum(notgpled,na.rm=TRUE)) %>% 
+  ungroup()
+
+#add files
```
```suggestion
# Combine Scottish Health Boards All Reasons data frames
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+Tabs136<-
+  bind_rows(ScotHBlaallreasonsalldelaystotal,ScotHBlaallreasonsincHSCPatFamtotal,ScotHBlaallreasonexcHSCPatFamtotal)
+
+arrange(Tabs136,areaname,age_grp,reas1) # arrange data in order for tables
```
See "Style Guide" section in general feedback comments.
```suggestion
Tabs136 <- bind_rows(ScotHBlaallreasonsalldelaystotal, ScotHBlaallreasonsincHSCPatFamtotal, ScotHBlaallreasonexcHSCPatFamtotal) %>%
    arrange(areaname, age_grp, reas1)
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+arrange(Tabs136,areaname,age_grp,reas1) # arrange data in order for tables
+
+
+#write_sav(Tabs136,paste0(filepath,"Tabs 1 3 6_R.sav"))
```
See "Saving (SPSS) Files Out" section in general feedback comments.
```suggestion
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+#  group_by(year,monthflag,level,areaname,age_grp,reason_grp_high_level) %>% 
+#  summarise(mean_delay=mean(delay_at_census),
+#            median_delay=median(delay_at_census)) %>% 
+#  ungroup()
```
This is all commented out. If it's not needed, it should be deleted. 
```suggestion
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+#            median_delay=median(delay_at_census)) %>% 
+#  ungroup()
+
+###12. Now get detailed reason breakdown ----
```
```suggestion
### 8 - Detailed reason breakdown ----
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+#  ungroup()
+
+###12. Now get detailed reason breakdown ----
+#get main census file ( datafile2)
```
```suggestion
# Get main census file (datafile2)
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+
+datafile5$level<-"1"
+datafile5$areaname<-"Scotland"
+datafile5$age_grp<-"All"
```
See "Incremental Data-Frame Names" section in general feedback.
```suggestion
datafile %>% mutate(level = "1", areaname = "Scotland", age_grp = "All")
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+datafile5$level<-"1"
+datafile5$areaname<-"Scotland"
+datafile5$age_grp<-"All"
+table(datafile5$REASONFORDELAY) # check REASONFORDELAY
```
See "Exploration/Checking Statements" section in general feedback.
```suggestion
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+#then update reas1 with individual delay codes in delayreason
+
+#datafile5<-datafile5 %>% mutate(REASONFORDELAY=
+#                                 if_else(REASONFORDELAY==9,dd_code_2,REASONFORDELAY))
```
See "Incremental Data-Frame Names" section in general feedback. It's also commented out so presumably not used?
```suggestion
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+            gpled=sum(gpled,na.rm=TRUE),notgpled=sum(notgpled,na.rm=TRUE)) %>% 
+  ungroup()
+
+# datafile5a<-datafile2
```
See "Incremental Data-Frame Names" section in general feedback. It's also commented out so presumably not used?
```suggestion
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+
+
+#match files ( excluding files with mean and media - no longer needed )
+
```
This is all commented out. If it's not needed, it should be deleted. 
```suggestion
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+datafile6<- bind_rows(Tabs136,Scotlandindreasons)
+
+#datafile6 <- left_join(Tabs136, Scotlandindreasons,
+#                       by = c(("year" = "year"), ("monthflag"="monthflag"),("level"="level"),("areaname"="areaname"), ("age_grp"="age_grp"),("reason_grp_high_level"="reason_grp_high_level")))
```
This is all commented out. If it's not needed, it should be deleted. 
```suggestion
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+#match files ( excluding files with mean and media - no longer needed )
+
+
+datafile6<- bind_rows(Tabs136,Scotlandindreasons)
```
See "Style Guide" and "Incremental Data-Frame Names" sections in general feedback. This is actually creating a new data-frame, however, so maybe using a more descriptive name would be useful?
```suggestion
datafile6 <- bind_rows(Tabs136, Scotlandindreasons)
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+
+
+
+datafile7<- bind_rows(datafile6,Scotlandindreasons)
```
See "Style Guide" and "Incremental Data-Frame Names" sections in general feedback. This is actually creating a new data-frame, however, so maybe using a more descriptive name would be useful?
```suggestion
datafile7 <- bind_rows(datafile6, Scotlandindreasons)
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+#                                 reason_grp_high_level%!in%c("All","Health and Social Care Reasons",
+#                                 "Code 9","All Delays excl.Code 9","Patient/Carer/Family-related reasons",0,notgpled)))
+
+#Note: used datafile6 as the creation of datafile7 doesn't add anything new - PMc: check .
```
This is all commented out. If it's not needed, it should be deleted.
```suggestion
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+
+#Note: used datafile6 as the creation of datafile7 doesn't add anything new - PMc: check .
+
+### 13. Save out census file ----
```
```suggestion
### 9 - Save out census file ----
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+
+### 13. Save out census file ----
+
+write_sav(datafile6,paste0(filepath,"All census data.sav"))
```
See "Saving (SPSS) Files Out" section in general feedback. In this case, however, I'm thinking there may be future dependencies on this file?
```suggestion
```

alice-hannah replied:
I think the output from this script is used only to populate the excel tables so don't think we need a sav version of any files.

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+
+write_sav(datafile6,paste0(filepath,"All census data.sav"))
+
+write.xlsx(datafile6,paste0(filepath,"All census data.xlsx"))
```
See "Style Guide" section in general feedback.
```suggestion
write.xlsx(datafile6, here::here("outputs", "All census data.xlsx"))
```

alice-hannah replied:
I'd just write this out as a csv using `readr::write_csv` to keep things simple. Also, I don't think an outputs folder is created in the setup-environment script. I'd suggest we add a bit of code to create this and a dated folder within.

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+
+write.xlsx(datafile6,paste0(filepath,"All census data.xlsx"))
+
+table(datafile6$areaname)
```
See "Exploration/Checking Statements" section in general feedback.
```suggestion
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+
+################################################################################
+### 14. bed days data ----
+################################################################################
```
```suggestion
### 10 - Bed days data ----
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+
+#Get filespath+main bed days file.SAV
+
+#datafile11<-read_spss(paste0(filepath,"main bed days file.sav"))
```
This is all commented out. If it's not needed, it should be deleted.
```suggestion
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+#datafile11<-read_spss(paste0(filepath,"main bed days file.sav"))
+
+
+datafile11<-datafile %>% mutate(Healthboard=substring(Healthboard, 5)) # change nhs_board and remove the 'NHS_'
```
See "Style Guide" and "Incremental Data-Frame Names" sections in general feedback.
```suggestion
datafile %>% mutate(Healthboard = substring(Healthboard, 5)) # change nhs_board and remove the 'NHS_'
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+  ungroup()
+
+
+#write_sav(labeddaysdiffage_grp,paste0(filepath,"la bed days diff age_grp_R.sav")) # save out file
```
See "Saving (SPSS) Files Out" section in general feedback.
```suggestion
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+  ungroup()
+
+
+#write_sav(Scotbeddaysdiffage_grp,paste0(filepath,"Scot bed days diff age_grp_R.sav")) # save out file
```
See "Saving (SPSS) Files Out" section in general feedback.
```suggestion
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+
+# add files
+
+Scotlandhbbeddays<-rbind(Scotbeddaysdiffage_grp,hbbeddaysdiffage_grp)
```
See "Style Guide" section in general feedback. 
You use `bind_rows()` throughout but move to `rbind()` here. `bind_rows()` as a `dplyr` function should replace `rbind()` here.
```suggestion
Scotlandhbbeddays <- bind_rows(Scotbeddaysdiffage_grp, hbbeddaysdiffage_grp)
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+  ungroup()
+
+#add files
+Scotlandhbbeddaysallagesallreasons<-rbind(Scotlandhbbeddays,Scotbeddaysdiffage_grp,hbbeddaysdiffage_grp)
```
See "Style Guide" section in general feedback.
You use bind_rows() throughout but move to rbind() here. bind_rows() as a dplyr function should replace rbind() here.
```suggestion
Scotlandhbbeddaysallagesallreasons <- bind_rows(Scotlandhbbeddays, Scotbeddaysdiffage_grp, hbbeddaysdiffage_grp)
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+#add files
+Scotlandhbbeddaysallagesallreasons<-rbind(Scotlandhbbeddays,Scotbeddaysdiffage_grp,hbbeddaysdiffage_grp)
+
+#write_sav(Scotandhbbeddaysallagesallreasons,paste0(filepath,"Scot and hb bed days_R.sav")) # save out file
```
See "Saving (SPSS) Files Out" section in general feedback.
```suggestion
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+
+Scotlandhbbeddaysallagegrpsallreasons<-rbind(ScotlandhbAllage_grpsbeddays,Scotlandhbbeddaysallagesallreasons)
+
+#write_sav(Scotlandhbbeddaysallagegrpsallreasons,paste0(filepath,"Scot and hb bed days all age_grps all reasons.sav")) # save out file
```
See "Saving (SPSS) Files Out" section in general feedback.
```suggestion
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+  summarise(OBDs=sum(OBDs,na.rm=TRUE)) %>% 
+  ungroup()
+
+Scotlandhbbeddaysallagegrpsallreasons<-rbind(ScotlandhbAllage_grpsbeddays,Scotlandhbbeddaysallagesallreasons)
```
See "Style Guide" section in general feedback.
You use bind_rows() throughout but move to rbind() here. bind_rows() as a dplyr function should replace rbind() here.
```suggestion
Scotlandhbbeddaysallagegrpsallreasons <- bind_rows(ScotlandhbAllage_grpsbeddays, Scotlandhbbeddaysallagesallreasons)
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+
+#write_sav(Scotlandhbbeddaysallagegrpsallreasons,paste0(filepath,"Scot and hb bed days all age_grps all reasons.sav")) # save out file
+
+### 15. Bed days LAs ----
```
```suggestion
### 11 - LA Bed days ----
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+### 15. Bed days LAs ----
+#use la bed days diff age file 
+datafile17<-labeddaysdiffage_grp %>% mutate(age_grp="All")
+table(datafile17$OBDs)
```
See "Exploration/Checking Statements" section in general feedback.
```suggestion
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+  summarise(OBDs=sum(OBDs,na.rm=TRUE)) %>% 
+  ungroup()
+
+datafile18<-rbind(labeddaysallage_grps,labeddaysdiffage_grp)
```
See "Style Guide" section in general feedback.
You use bind_rows() throughout but move to rbind() here. bind_rows() as a dplyr function should replace rbind() here.
```suggestion
datafile18 <- bind_rows(labeddaysallage_grps, labeddaysdiffage_grp)
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+
+datafile18<-rbind(labeddaysallage_grps,labeddaysdiffage_grp)
+
+#write_sav(datafile18,paste0(filepath,"la bed days.sav")) # save out file
```
See "Saving (SPSS) Files Out" section in general feedback.
```suggestion
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+  summarise(OBDs=sum(OBDs,na.rm=TRUE)) %>% 
+  ungroup()
+
+datafile20<-rbind(labeddaysallreason_grp_high_level,datafile18)
```
See "Style Guide" section in general feedback.
You use bind_rows() throughout but move to rbind() here. bind_rows() as a dplyr function should replace rbind() here.
```suggestion
datafile20 <- bind_rows(labeddaysallreason_grp_high_level, datafile18)
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+
+datafile20<-rbind(labeddaysallreason_grp_high_level,datafile18)
+
+#write_sav(datafile20,paste0(filepath,"la bed days all age_grps all reasons.sav")) # save out file
```
See "Saving (SPSS) Files Out" section in general feedback.
```suggestion
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+
+#write_sav(datafile20,paste0(filepath,"la bed days all age_grps all reasons.sav")) # save out file
+
+#### 16. Bad Days HB ----
```
```suggestion
### 12 - HB Bed days ----
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+#write_sav(datafile20,paste0(filepath,"la bed days all age_grps all reasons.sav")) # save out file
+
+#### 16. Bad Days HB ----
+#match in to hb data sheet template
+hbbedstemplate<-read.csv(paste0(filepath2,"hb_template.csv"))
+
+hbbedstemplate <- hbbedstemplate %>% 
+  rename(Healthboard=hbname,
+         age_grp = age)
+
+
+hbbedstemplate<-hbbedstemplate%>% mutate(Healthboard=toupper(Healthboard),age_grp=toupper(age_grp),reas1=toupper(reas1))
```
Alice has asked previously about why `hbbedstemplate` is necessary, I'm not sure if this has been answered. Changes with this will obviously affect later code but it's worth discussing this. I'll quote her comment:

> Unsure of why we need to use a template. There seems to be quite a lot of changes made once it's read in and so I'm wondering if we need it at all?
> I'm wondering if you've used it to make sure you have a row for every combination of the three variables? If so, you could try using tidyr::complete() to do this without the need for a template.

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+
+arrange(datafile21,Healthboard,age_grp,reas1) # arrange data in order for tables
+
+#write_sav(datafile21,paste0(filepath,"hb bed days data sheet minus standard.sav")) # save out file
```
See "Saving (SPSS) Files Out" section in general feedback.
```suggestion
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+arrange(datafile21,Healthboard,age_grp,reas1) # arrange data in order for tables
+
+#write_sav(datafile21,paste0(filepath,"hb bed days data sheet minus standard.sav")) # save out file
+datafile21<-datafile21%>% mutate(Healthboard=toupper(Healthboard),age_grp=toupper(age_grp),reas1=toupper(reas1))
```
See "Pipes" and "Incremental Data-Frame Names" sections in general feedback.
Why are we changing variable names to be upper case?
```suggestion
datafile %<>% mutate(Healthboard = toupper(Healthboard), age_grp = toupper(age_grp), reas1 = toupper(reas1))
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+  ungroup()
+
+
+#write_sav(hbbeddaysdiffage_grp,paste0(filepath,"hb bed days diff age_grp_R.sav")) # save out file
```
See "Saving (SPSS) Files Out" section in general feedback.
```suggestion
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+datafile21<-datafile21%>% mutate(Healthboard=toupper(Healthboard),age_grp=toupper(age_grp),reas1=toupper(reas1))
+
+#Need to ensure there is a total for standard delays (HSC/PFR added together). This is due to the format of the publication data sheet.
+#datafile22a<-filter(datafile21,reason_grp_high_level!="ALL") 
```
This is all commented out. If it's not needed, it should be deleted.
```suggestion
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+#Rename reason_grp_high_level as Standard where there isn't a Code 9
+datafile22<-datafile21%>% mutate(reas1=
+                                   if_else(reas1%in%c("HEALTH AND SOCIAL CARE REASONS","PATIENT/CARER/FAMILY-RELATED REASONS"),"STANDARD",reas1))
+table(datafile22$reas1)
```
See "Exploration/Checking Statements" section in general feedback.
```suggestion
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+datafile27<-arrange(datafile27,datafile27$Healthboard,datafile27$age_grp,datafile27$reas1)
+
+
+### 17. Save out bed days HB file ----
```
```suggestion
### 13 - Save out bed days HB file ----
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+
+### 17. Save out bed days HB file ----
+
+write.xlsx(datafile27,paste0(filepath,"HB bed days data sheet_R.xlsx"))
```
```suggestion
write.xlsx(datafile27, here::here("outputs", "HB bed days data sheet_R.xlsx"))
```

alice-hannah replied:
Same comment as I made above re census file. I'd use `write_csv` and keep in mind we need to edit setup-environment script to create outputs folder. I'd also suggest we review the names of the resulting file as this seems a bit unnecessarily long? I'd call it something like - `2020-07_bed-days.csv`. And `2020-07_census.csv`.

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+
+#########################################
+### 18. LA bed days  ----
+#########################################
```
```suggestion
### 14 - LA bed days ----
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+  rename(LocalAuthorityArea = LA,
+         age_grp = age)
+
+labedstemplate<-labedstemplate%>% mutate(LocalAuthorityArea=toupper(LocalAuthorityArea),age_grp=toupper(age_grp),reas1=toupper(reas1))
+datafile20<-datafile20%>% mutate(LocalAuthorityArea=toupper(LocalAuthorityArea),age_grp=toupper(age_grp),reas1=toupper(reas1))
```
Why are we changing variable names to be upper case?

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+labedstemplate<-labedstemplate%>% mutate(LocalAuthorityArea=toupper(LocalAuthorityArea),age_grp=toupper(age_grp),reas1=toupper(reas1))
+datafile20<-datafile20%>% mutate(LocalAuthorityArea=toupper(LocalAuthorityArea),age_grp=toupper(age_grp),reas1=toupper(reas1))
+
+##################################################################
```
```suggestion
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+
+arrange(labeddaysdatasheetminusstandard,LocalAuthorityArea,age_grp,reas1) # arrange data in order for tables
+
+#write_sav(labeddaysdatasheetminusstandard,paste0(filepath,"la bed days data sheet minus standard.sav")) # save out file
```
See "Saving (SPSS) Files Out" section in general feedback.
```suggestion
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+#Rename reason_grp_high_level as Standard where there isn't a Code 9
+datafile32<-datafile32a%>% mutate(reas1=
+                                    if_else(reas1%in%c("HEALTH AND SOCIAL CARE REASONS","PATIENT/CARER/FAMILY-RELATED REASONS"),"STANDARD","CODE 9"))
+table(datafile32$reas1)
```
See "Exploration/Checking Statements" section in general feedback.
```suggestion
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+labeddaysdatasheetminusstandard <- left_join(datafile20,labedstemplate,
+                                             by = c(("LocalAuthorityArea" = "LocalAuthorityArea"), ("age_grp"="age_grp"),("reas1"="reas1")))
+
+arrange(labeddaysdatasheetminusstandard,LocalAuthorityArea,age_grp,reas1) # arrange data in order for tables
```
The `left_join()` function needs to have the common variables listed. As they are the same name in both of these data frames, they don't need to be listed to equal each other.
The `arrange()` function needs a data frame input as the first argument, or if using the pipe, to be part of that chain. As it stands, `arrange()` isn't doing anything here. Let's attach it to the previous code.
```suggestion
labeddaysdatasheetminusstandard <- left_join(datafile20, labedstemplate, 
    by = c("LocalAuthorityArea", "age_grp", "reas1")) %>% 
    arrange(labeddaysdatasheetminusstandard, LocalAuthorityArea, age_grp, reas1)
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+
+#amend variables as necessary in order for matching to work
+
+datafile21 <- left_join(Scotlandhbbeddaysallagegrpsallreasons,hbbedstemplate,
+                        by = c(("Healthboard" = "Healthboard"), ("age_grp"="age_grp"),("reas1"="reas1")))
+
+arrange(datafile21,Healthboard,age_grp,reas1) # arrange data in order for tables
```
The left_join() function needs to have the common variables listed. As they are the same name in both of these data frames, they don't need to be listed to equal each other.
The arrange() function needs a data frame input as the first argument, or if using the pipe, to be part of that chain. Let's attach it to the previous code.
```suggestion
datafile21 <- left_join(Scotlandhbbeddaysallagegrpsallreasons, hbbedstemplate, 
    by = c("Healthboard", "age_grp", "reas1")) %>%
    arrange(Healthboard, age_grp, reas1)
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+  summarise(OBDs=sum(OBDs,na.rm=TRUE)) %>% 
+  ungroup()
+
+#datafile24<-datafile24%>%rename(OBDs2=obds_in_month)
```
This is all commented out. If it's not needed, it should be deleted.
```suggestion
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+lastandardandothers<-lastandardandothers%>% mutate(LocalAuthorityArea=toupper(LocalAuthorityArea),age_grp=toupper(age_grp),reas1=toupper(reas1))
+
+
+arrange(lastandardandothers,LocalAuthorityArea,age_grp,reas1) # issue here is that the rows with zeros don't appear so have to match to output file
```
See "Style Guide" and "Pipes" sections in general feedback.
Why are we changing these variables to be upper case?
What do you mean "rows with zeros don't appear"?
```suggestion
lastandardandothers <- bind_rows(lastandard, labeddaysdatasheetminusstandard) %>%
    mutate(LocalAuthorityArea = toupper(LocalAuthorityArea), age_grp = toupper(age_grp), reas1 = toupper(reas1)) %>%
    arrange(LocalAuthorityArea, age_grp, reas1) # issue here is that the rows with zeros don't appear so have to match to output file
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+
+arrange(lastandardandothers,LocalAuthorityArea,age_grp,reas1) # issue here is that the rows with zeros don't appear so have to match to output file
+
+#datafile26<-read.csv(paste0(filepath2,"hb_template.csv"))
+#bhbbedstepmplate<-bhbbedstepmplate%>% mutate(nhs_board=toupper(nhs_board),age_grp=toupper(age_grp),reason_grp_high_level=toupper(reason_grp_high_level))
```
This is all commented out. If it's not needed, it should be deleted.
```suggestion
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+
+
+
+laallvariations<-arrange(laallvariations,laallvariations$LocalAuthorityArea,laallvariations$age_grp,laallvariations$reas1)
```
See "Pipes" section in general feedback. I've also left comments above about the arguments required for `left_join()`. 
```suggestion
laallvariations <- left_join(labedstemplate, lastandardandothers, 
    by = c("LocalAuthorityArea", "age_grp", "reas1")) %>%
    # replace obds in correct column for all reason groups
    mutate(OBDs = if_else(is.na(OBDs), 0L, OBDs)) %>%
    arrange(LocalAuthorityArea, age_grp, reas1)
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+
+laallvariations<-arrange(laallvariations,laallvariations$LocalAuthorityArea,laallvariations$age_grp,laallvariations$reas1)
+
+### 19. Save out LA file ----
```
```suggestion
### 15 - Save out LA file ----
```

rmccreath commented on code/03_census_and_bed_days.R:
```diff
@@ -0,0 +1,868 @@
+
+### 19. Save out LA file ----
+
+write.xlsx(laallvariations,paste0(filepath,"LA bed days data sheet_R.xlsx"))
```
```suggestion
write.xlsx(laallvariations, here::here("outputs", "LA bed days data sheet_R.xlsx"))
```

rmccreath reviewed (changes requested):
Good going, @petermcclurg 
I've completed the code review as requested for `03_census_and_bed_days.R`. I think you've submitted `01_validation` by mistake? 
I've tried to be thorough without being difficult to follow (or a total pain to you) so there's inline comments (and some questions) with as much new code suggestions as possible, this will make it easier to either accept the changes or at least see where my thought process was. To avoid duplication of comments, I've added some general feedback below and refer to this inline with your code throughout. 
It'd be great if you could review this and I would suggest putting any other development on pause and going through this line-by-line from the top. I'm happy to do some pair-programming or just have a chat about any of it when you'd like. I know you're only back on Monday after a well-deserved break so I'll you get settled back in first.

**Style Guide**
I've made changes as part of other code suggestions but not all, so there is still some work to refine the code. The high-level themes to follow are:
- file names should be lowercase
- there should be a space either side of the assignment and comparison operators, e.g. `<-`, `=` and `!=`
- there should be a space after commas, not before, e.g. commas (`,`) between function arguments
- see [style.tidyverse.org](style.tidyverse.org) for more
- utilise Tidyverse functions where possible and remain consistent throughout the script

**Pipes**
Generally, we use pipes to make code more readable. They're used to chain commands, there are a few more opportunities to utilise the pipe in the script.
Additionally, while this isn't actually recommended as part of the above style guide, it is recommended in PHS - when assigning a change to a dataset, use `dataset %<>% function()` rather than `dataset <- dataset %>% function()`. They are both good code but using `%<>%` makes the code more readable. `%<>%` comes from the `magrittr` package which is already loaded in `setup_environment`.

**Exploration/Checking Statements**
There are a few data exploration/checking statements in here that are commented out (e.g. `sum()` and `table()`). Are these necessary or just as part of building this script? There are better ways to incorporate checking that mean running this file wouldn't turn into a manual process.

**Saving (SPSS) Files Out**
These tend to be commented out and I think might be coming from a "direct" translation from the original SPSS. If they're commented out, are they needed? Also, we shouldn't be exporting to SPSS (.sav) formats if we can possibly avoid it.

**Incremental Data-Frame Names**
Throughout the script, there are new data frames created by incrementing `1` to the name. I think this is coming from a "direct" translation from the original SPSS and in many cases isn't required? Where it gets to `dataframe6` the data is different. This kind of refers to a Style Guide comment, but variable names (including the name of data frames) should be descriptive. I've attempted to align the naming of data-frame in my suggestions but this will need reviewed (or ignored).

**Code Needs Refactoring**
There are some sections that presumably come from a direct translation from SPSS, these may have dependencies with other processes for the team. As such, I haven't included recommendations to alter these but they should be flagged for future code refactoring. This is likely to be affected by *Incremental Data-Frame Names* above.

**Incorrect Variable Names**
There are cases where the variable names aren't right, either from the previous changes that were made or because of how `02_create_scotland_file` sets them up. I've highlighted some of these but you'll need to check over the whole file again. Remember to change all instances at once you can: 
1. Find an instance of the variable to be changed (you can use `ctrl` + `f` for this part).
2. Highlight the variable instance (double click).
3. Hold `ctrl` + `shift` + `alt` + `M` - this will highlight all instances of that variable in scope.
4. Type the new name for that variable.

petermcclurg committed:
Changes made to be run and checked Tuesday AM

petermcclurg committed:
Numbering updated on previous version

petermcclurg committed:
Run census and bed days to line 167 where error appears. Also had to manually read in file as here doesn't work for me. Please advise.

petermcclurg committed:
merge with master for create scotland file

Merge branch 'master' into peter

# Please enter a commit message to explain why this merge is necessary,
# especially if it merges an updated upstream into a topic branch.
#
# Lines starting with '#' will be ignored, and an empty message aborts
# the commit.

rmccreath commented on code/02_create_scotland_file.R:
```diff
@@ -16,6 +16,8 @@
 ### 0 - Load setup environment and functions ----
 
 source(here::here("code", "00_setup_environment.R"))
+remotes::install_github("Health-SocialCare-Scotland/phsmethods",
+                        upgrade = "never")
```
`phsmethods` is brought in with `setup_envinronment.R`
```suggestion
```

petermcclurg commented:
HI Russell,

I think we installed this manually  as RStudio version 3.6.1 didn’t have phsmethods.

Should this have been done before running the setup_environment.R?

Peter

From: Russell McCreath <notifications@github.com>
Sent: 13 October 2020 11:46
To: Public-Health-Scotland/delayed-discharges-monthly <delayed-discharges-monthly@noreply.github.com>
Cc: petermcclurg <petermcclurg@nhs.net>; Mention <mention@noreply.github.com>
Subject: Re: [Public-Health-Scotland/delayed-discharges-monthly] Peter (#4)


@rmccreath commented on this pull request.

________________________________

In code/02_create_scotland_file.R<https://github.com/Public-Health-Scotland/delayed-discharges-monthly/pull/4#discussion_r503850935>:

> +remotes::install_github("Health-SocialCare-Scotland/phsmethods",



+                        upgrade = "never")



phsmethods is brought in with setup_envinronment.R
⬇️ Suggested change

-remotes::install_github("Health-SocialCare-Scotland/phsmethods",



-                        upgrade = "never")



—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/Public-Health-Scotland/delayed-discharges-monthly/pull/4#pullrequestreview-507303689>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ANGSC6EAG7XO22BVXTUYQATSKQVWRANCNFSM4QHFKBWA>.





--------------------------------------------------------------------------------------------------------------------------------------------------------
This email is intended for the named recipient only. If you have received it by mistake,
please (i) contact the sender by email reply; (ii) delete the email from your system; .
and (iii) do not copy the email or disclose its contents to anyone.

--------------------------------------------------------------------------------------------------------------------------------------------------------

rmccreath commented:
The installation of packages is per user, so you may have needed to install it for your profile for when you started to use 3.6.1. It doesn’t need to be done every time the script is run, it only needs to be loaded, i.e. with `library(phsmethods)`. Assuming you have the package installed for yourself now, you can delete this and `setup_environment` will continue to load the package as required.

From: petermcclurg <notifications@github.com>
Reply to: Public-Health-Scotland/delayed-discharges-monthly <reply@reply.github.com>
Date: Tuesday, 13 October 2020 at 13:07
To: Public-Health-Scotland/delayed-discharges-monthly <delayed-discharges-monthly@noreply.github.com>
Cc: Russell McCreath <Russell.Mccreath@phs.scot>, Mention <mention@noreply.github.com>
Subject: Re: [Public-Health-Scotland/delayed-discharges-monthly] Peter (#4)

HI Russell,

I think we installed this manually as RStudio version 3.6.1 didn’t have phsmethods.

Should this have been done before running the setup_environment.R?

Peter

From: Russell McCreath <notifications@github.com>
Sent: 13 October 2020 11:46
To: Public-Health-Scotland/delayed-discharges-monthly <delayed-discharges-monthly@noreply.github.com>
Cc: petermcclurg <petermcclurg@nhs.net>; Mention <mention@noreply.github.com>
Subject: Re: [Public-Health-Scotland/delayed-discharges-monthly] Peter (#4)


@rmccreath commented on this pull request.

________________________________

In code/02_create_scotland_file.R<https://github.com/Public-Health-Scotland/delayed-discharges-monthly/pull/4#discussion_r503850935>:

> +remotes::install_github("Health-SocialCare-Scotland/phsmethods",



+ upgrade = "never")



phsmethods is brought in with setup_envinronment.R
⬇️ Suggested change

-remotes::install_github("Health-SocialCare-Scotland/phsmethods",



- upgrade = "never")



—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/Public-Health-Scotland/delayed-discharges-monthly/pull/4#pullrequestreview-507303689>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ANGSC6EAG7XO22BVXTUYQATSKQVWRANCNFSM4QHFKBWA>.





--------------------------------------------------------------------------------------------------------------------------------------------------------
This email is intended for the named recipient only. If you have received it by mistake,
please (i) contact the sender by email reply; (ii) delete the email from your system; .
and (iii) do not copy the email or disclose its contents to anyone.

--------------------------------------------------------------------------------------------------------------------------------------------------------


—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/Public-Health-Scotland/delayed-discharges-monthly/pull/4#issuecomment-707694102>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AIDECFSONILGJJJIVBQDHQDSKQ7GPANCNFSM4QHFKBWA>.





--------------------------------------------------------------------------------------------------------------------------------------------------------
This email is intended for the named recipient only. If you have received it by mistake,
please (i) contact the sender by email reply; (ii) delete the email from your system; .
and (iii) do not copy the email or disclose its contents to anyone.

--------------------------------------------------------------------------------------------------------------------------------------------------------
